Repository: https://github.com/washingtonjustin02smc/CS82A-repository

In [1]:
import pandas as pd

df = pd.read_csv("allbacks.csv")
df

 rownames  volume  area  weight cover
        1     885   382     800    hb
        2    1016   468     950    hb
        3    1125   387    1050    hb
        4     239   371     350    hb
        5     701   371     750    hb
        6     641   367     600    hb
        7    1228   396    1075    hb
        8     412     0     250    pb
        9     953     0     700    pb
       10     929     0     650    pb
       11    1492     0     975    pb
       12     419     0     350    pb
       13    1010     0     950    pb
       14     595     0     425    pb
       15    1034     0     725    pb

In [2]:
ordered_covers = sorted(df["cover"].tolist())
median_cover = ordered_covers[len(ordered_covers) // 2]
median_cover

'pb'

**Median cover type:** The median cover type is paperback which means the middle cover type in the ordered data is paperback.

In [3]:
mean_weight = df["weight"].mean()
mean_weight

706.6666666666666

**Mean weight:** The average weight of all the books is **706.67 grams**, which represents the typical weight of a book in this dataset.

In [4]:
mean_by_cover = df.groupby("cover")["weight"].mean()
mean_by_cover

cover
hb    796.428571
pb    628.125000
Name: weight, dtype: float64

**Hardback vs. paperback:** Hardback books have an average weight of **796.43 grams**, paperback books average **628.13 grams**. That means that hardback books tend to weigh more than paperback books.

**By-hand check:**

Book weights: 800 + 950 + 1050 + 350 + 750 + 600 + 1075 + 250 + 700 + 650 + 975 + 350 + 950 + 425 + 725 = **10,600 grams**

10,600 / 15 = **706.67 grams**

This matches the calculated mean weight.